File: Smart-Parking-Backend/src/main/java/com/smartparking/service/SmsService.java
package com.smartparking.service;

import org.springframework.stereotype.Service;

@Service
public class SmsService {

    public Boolean sendSms(String phoneNumber, String message) {
        try {
            if (phoneNumber == null || phoneNumber.isEmpty()) {
                return false;
            }
            
            System.out.println("SMS Sent to: " + phoneNumber);
            System.out.println("Message: " + message);
            
            return true;
        } catch (Exception e) {
            System.out.println("Failed to send SMS: " + e.getMessage());
            return false;
        }
    }

    public Boolean sendBookingConfirmationSms(String phoneNumber, Long bookingId, String slotNumber) {
        String message = "Your parking booking is confirmed. Booking ID: " + bookingId + 
                        ", Slot: " + slotNumber + ". Thank you!";
        return sendSms(phoneNumber, message);
    }

    public Boolean sendPaymentAlertSms(String phoneNumber, Double amount) {
        String message = "Payment of $" + amount + " has been received. Thank you for using Smart Parking!";
        return sendSms(phoneNumber, message);
    }

    public Boolean sendParkingReminderSms(String phoneNumber, String duration) {
        String message = "Your parking session will end in " + duration + ". Please plan accordingly.";
        return sendSms(phoneNumber, message);
    }
}

File: Parking-Frontend/src/services/smsService.js
import api from "../api/axios";

export const smsService = {
  /**
   * Send SMS (admin only)
   * @param {Object} data - {phoneNumber, message, type}
   */
  sendSMS: async (data) => {
    const response = await api.post("/sms/send", data);
    return response.data;
  },

  /**
   * Send booking confirmation SMS (admin only)
   * @param {Object} params - {phoneNumber, bookingId, slotNumber}
   */
  sendBookingConfirmationSMS: async (params) => {
    const queryParams = new URLSearchParams();
    queryParams.append("phoneNumber", params.phoneNumber);
    queryParams.append("bookingId", params.bookingId);
    queryParams.append("slotNumber", params.slotNumber);

    const response = await api.post(
      `/sms/booking-confirmation?${queryParams}`
    );
    return response.data;
  },

  /**
   * Send payment alert SMS (admin only)
   * @param {Object} params - {phoneNumber, amount}
   */
  sendPaymentAlertSMS: async (params) => {
    const queryParams = new URLSearchParams();
    queryParams.append("phoneNumber", params.phoneNumber);
    queryParams.append("amount", params.amount);

    const response = await api.post(`/sms/payment-alert?${queryParams}`);
    return response.data;
  },

  /**
   * Send parking reminder SMS (admin only)
   * @param {Object} params - {phoneNumber, duration}
   */
  sendParkingReminderSMS: async (params) => {
    const queryParams = new URLSearchParams();
    queryParams.append("phoneNumber", params.phoneNumber);
    queryParams.append("duration", params.duration);

    const response = await api.post(`/sms/parking-reminder?${queryParams}`);
    return response.data;
  },
};
